core/parser.cpp

```diff
@@ -1186,9 +1186,42 @@ namespace {
 
                     Token op = pop();
                     if (op.kind == Token::BRACKET_L) {
-                        AST *index = parse(MAX_PRECEDENCE);
+                        const int max = 2;
+                        bool isSlice = true;
+                        AST *first = nullptr;
+                        AST *last = nullptr;
+
+                        for (int i = 0; i < max; i++) {
```

Comment:
Ideally the code should read like the BNF:

```
e ::= ...
    | e [ e [ ':' [ [ e ] [ ':' [ e ] ] ] ] ]
```
I think that's the right syntax anyway :)